pkg/config/viper.go

```diff
@@ -335,8 +335,10 @@ func (c *safeConfig) UnmarshalExact(rawVal interface{}) error {
 // ReadInConfig wraps Viper for concurrent access
 func (c *safeConfig) ReadInConfig() error {
 	c.Lock()
-	defer c.Unlock()
-	return c.Viper.ReadInConfig()
+	err := c.Viper.ReadInConfig()
+	c.Unlock()
+	promoteExperimentalOTLP(c) // TODO(gbbr): remove after 7.35.0 is released
```

Comment:
Should we move this to `SetupOTLP`? Or is there some problem with that?

Reply:
`SetupOTLP` happens before reading the file and constructing the config, so it's not possible to do it at that point. It is simply a preparation step. Also, it was previously in `Load` and that was incorrect. Some tests use `ReadInConfig` and they were failing. Config should always be read correctly from file.